python/mxnet/mxfeatures.py

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=not-an-iterable
+
+"""runtime detection of compile time features in the native library"""
+
+import ctypes
+import enum
+from .base import _LIB, check_call, mx_uint
+
+feature_names = [
```

Comment:
Curious: any automated way to keep the names/enum in sync?

Reply:
Not that I know of.

Reply:
Where was that list taken from (e.g. Makefile?)? Maybe we could have a script that we run in the sanity-check stage that ensures that this list of features is in sync with the source and otherwise prompt the contributor to add the feature.

@ThomasDelteil did the same for tutorials

Reply:
Offering it as CAPI would make it easier to maintain. You can use ctypes to query this CAPI for the list of available features.

Reply:
@szha You mean something like `const char* get_feature_name(int feature)`?

Reply:
You may refer to how the operators are registered in mxnet. https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/base.py#L578